api/views.py

```diff
@@ -1,111 +1,100 @@
+    amount if user has enabled this in settings.
     """
-    model = Commission
-    permission_classes = [IsAuthenticated]
-    serializer_class = CreateCommissionSerializer
-    queryset = Commission.objects.all()
+    serializer_class = CheckBalanceSerializer
 
-    def post(self, request, *args, **kwargs):
-        self.check_permissions(request)
-        serializer = self.get_serializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
+    def get(self, request, *args, **kwargs):
+        card_uuid = self.request.META.get('HTTP_CARD_NUMBER', None)
+        soci_bank_account: SociBankAccount = self._get_account_from_card_id(card_uuid=card_uuid)
+        serializer = self.get_serializer(soci_bank_account)
+        data = serializer.data
 
-        commission = serializer.save()
-        data = CommissionSerializer(commission).data
+        response_status = status.HTTP_200_OK
+        if not soci_bank_account.has_sufficient_funds:
+            response_status = status.HTTP_402_PAYMENT_REQUIRED
 
-        return Response(data, status.HTTP_201_CREATED)
+        return Response(data, status=response_status)
 
+    @staticmethod
+    def _get_account_from_card_id(card_uuid) -> SociBankAccount:
+        soci_bank_account = get_object_or_404(queryset=SociBankAccount.objects.all(), card_uuid=card_uuid)
 
-class UpdateCommissionView(generics.UpdateAPIView):
-    """
-    Updating an existing commission in the database.
-    """
-    model = Commission
-    permission_classes = [IsAuthenticated]
-    serializer_class = CreateCommissionSerializer
-    lookup_url_kwarg = 'commission_id'
-    queryset = Commission.objects.all()
+        return soci_bank_account
 
-    def patch(self, request, *args, **kwargs):
-        commission = self.get_object()
-        serializer = self.get_serializer(commission, data=request.data, partial=True)
-        serializer.is_valid(raise_exception=True)
-        serializer.save()
-
-        data = CommissionSerializer(commission).data
-
-        return Response(data, status=status.HTTP_201_CREATED)
 
-
-class DeleteCommissionView(generics.DestroyAPIView):
+class ChargeSociBankAccountView(generics.CreateAPIView):
     """
-    Deleting an existing commission in the database.
+    #### Charges the specified account, based on the provided RFID card number, \
+    for the price of the product connected to the provided SKU number.
     """
-    model = Commission
-    permission_classes = [IsAuthenticated]
-    serializer_class = CreateCommissionSerializer
-    lookup_url_kwarg = 'commission_id'
-    queryset = Commission.objects.all()
+    serializer_class = TransactionSerializer
 
-    def delete(self, request, *args, **kwargs):
-        commission = self.get_object()
-        commission.delete()
+    def post(self, request, *args, **kwargs):
+        card_uuid = self.request.META.get('HTTP_CARD_NUMBER', None)
```

Comment:
Is there any particular reason this data is read from a custom header?

Reply:
I didn't want to put the plaintext card id in the URL. We could discuss an alternative scheme for this when implementing the general API security feature?

Reply:
Yeah agreed. I believe a number of standards and "standard practices" suggest against using custom headers for data pertaining to a view itself (see for instance [here](https://www.oreilly.com/library/view/restful-web-services/9780596809140/ch01s13.html)), but I am not completely sure about that.